[4.0] version number

astridx opened the pull request:
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/26373.

### Summary of Changes
~~I changed the position of the module that displayed the version number to the bottom.~~ 
I changed the position of the module that displayed the version number. See https://github.com/joomla/joomla-cms/pull/26447#issuecomment-538655489

### Testing Instructions
1. Apply this patch.
2. Make a new installation (Have to update SQL in order for the patch to work).
3. See, that the version number is now ~~on the right side of the bottom~~  at the top right corner.


4. See, that no version number is displayed in the top menu.


### Documentation Changes Required
No

astridx committed:
version number

brianteeman commented on administrator/templates/atum/html/layouts/chromes/bottom-version.php:
```diff
@@ -0,0 +1,18 @@
+ * @copyright   Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ *
+ * Module chrome for rendering the module in a submenu
```
??

astridx replied:
@brianteeman Thank you Brian. I deleted the whole file, as we do not need it for this PR. I forgot this before.

astridx committed:
correction

Quy commented on administrator/modules/mod_version/tmpl/default.php:
```diff
@@ -12,14 +12,10 @@
-			<span aria-hidden="true"><?php echo $version; ?></span>
-		</div>
+<div class="col-md-12 ">
+	<div class="float-right ">
```
```suggestion
	<div class="float-right">
```

astridx replied:
@Quy Thank you. Actually, I wanted it that way. I think it's safer to set the space. So it can not happen so easily that when adding another class the two names are connected to one. But you are right. Otherwise this is not so in Joomla and everything should be consistent. I just corrected it.

astridx committed:
correction

astridx committed:
conflict

N6REJ commented:
TY TY TY TY TY,   I think I put an issue in for this...or maybe I just ranted

N6REJ commented:
I have tested this item :red_circle: unsuccessfully on 8e6f08bc2428a9d3640900f8207bb9ebafe05eac<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26447">issues.joomla.org/tracker/joomla-cms/26447</a>.</sub>

astridx commented:
@N6REJ Thank you for testing. Did you make a new installation of Joomla!?
What is the exact error message?

N6REJ commented:
If you made a "new" installation the patch wouldn't exist :person_with_blond_hair: 
I did 
-git reset --hard 
-git pull -f
-add patch
-npm ci
test.

There is no 'error' that was displayed.  If you look where the version WAS you'll see that its just messy text now.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26447">issues.joomla.org/tracker/joomla-cms/26447</a>.</sub>

astridx commented:
@N6REJ 
Please try:
1. Add the patch via patch tester.
2. Delete or rename the file configuration.php.
3. Run a new installation.

N6REJ commented:
I have tested this item :white_check_mark: successfully on 8e6f08bc2428a9d3640900f8207bb9ebafe05eac<br /><br />Must update SQL in order for the patch to work.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26447">issues.joomla.org/tracker/joomla-cms/26447</a>.</sub>

N6REJ commented:
> @N6REJ
> Please try:
> 
> 1. Add the patch via patch tester.
> 2. Delete or rename the file configuration.php.
> 3. Run a new installation.

how is doing a new installation going to keep the patch?

astridx commented:
@N6REJ 
That has overlapped. Great. Thank you very much for testing.

N6REJ commented:
might want to modify the testing instructions

alikon commented:
I have tested this item :white_check_mark: successfully on 8e6f08bc2428a9d3640900f8207bb9ebafe05eac<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26447">issues.joomla.org/tracker/joomla-cms/26447</a>.</sub>

alikon commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26447">issues.joomla.org/tracker/joomla-cms/26447</a>.</sub>

wilsonge commented:
I'm not convinced we want this? Part of the design of the new template is to remove the footer. Why is having the version in the header so totally wrong?

astridx commented:
I felt like @N6REJ . 
1. I also thought something had to be clicked here because the version looks like a button. 
2. I also believe that information that is not incidental should be displayed below. 
3. Plus, I think it's great to create the design the way most people expect it to. I did not do a big survey, but people around me are expecting the version number in the footer. 

Is there a reason why the footer was removed?

brianteeman commented:
I agree that currently it looks like its something you should click on
I agree that it probably doesn't need to be so prominent
I dont agree with this PR for the reason shown in your last screenshot

astridx commented:
So, you would prefer this:

![Articles   test   Administration](https://user-images.githubusercontent.com/9974686/66119743-94493380-e5d9-11e9-81bd-e9a1509e95ef.png)

astridx committed:
change position

N6REJ commented:
@astridx how about moving it to the status page?  Do we really need it on every page?

astridx commented:
@N6REJ It is already there:
![System Information   test   Administration(1)](https://user-images.githubusercontent.com/9974686/66154299-71407300-e61d-11e9-86e6-88e8633debcd.png)

brianteeman commented:
@astridx personally yes

@bear 
> do we really need it on every page?
You can make that comment about everything

N6REJ commented:
Here is where I meant but as you see it's already there... lets just remove it then!
![image](https://user-images.githubusercontent.com/1850089/66154562-5e10b180-e5e3-11e9-93c4-779121abb9c1.png)

brianteeman commented:
depends if you publish that module

N6REJ commented:
it's pre-published during install brian

chmst commented:
Did you do the test also on your smartphones?
The module adds a warning badge to the version number if the version ist not up to date. 
It was expected that people see this better in in the prominent place in the header. 

You can ask if it is necessary - but we want that every Joomla Version is always up to date. No casual user will search for the version number in the stats module or system information. 
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26447">issues.joomla.org/tracker/joomla-cms/26447</a>.</sub>

astridx committed:
correciton for small mobil and debug

astridx commented:
@chmst Thank you for checking.

For me it looks like this on the Smartphone:

![Articles   test   Administration(3)](https://user-images.githubusercontent.com/9974686/66229057-44ee2a80-e6e1-11e9-8006-07f9f11e6414.png)


But you are right. I did not test it correctly. If the screen width is less than 460px or debug mode is enabled, the version number disappears. That's why I gave the element with the version number a margin and a padding in the last commit. Now it looks like this:

![Articles   test   Administration(2)](https://user-images.githubusercontent.com/9974686/66229116-664f1680-e6e1-11e9-8660-d7849a6157b3.png)
![Articles   test   Administration(1)](https://user-images.githubusercontent.com/9974686/66229118-664f1680-e6e1-11e9-9603-df1d7750474e.png)
![Articles   test   Administration(4)](https://user-images.githubusercontent.com/9974686/66229132-6a7b3400-e6e1-11e9-9c69-0c2fad9926d9.png)

Without the version number we could perhaps position the buttons in the mobile foot menu directly and you would not have to open something extra.

What exactly do you mean with:
> The module adds a warning badge to the version number if the version ist not up to date.

I can not find a program code in the module that displays a warning if the version is outdated. Can you point me there please?

Without knowing the code, I wonder if the Quick-icon and the Joomla Updater update-hint are not enough to inform the user about an existing new version.
![Control Panel   test   Administration](https://user-images.githubusercontent.com/9974686/66229736-cbefd280-e6e2-11e9-98c6-173801afc2e9.png)

astridx committed:
Merge branch '4.0-dev' of https://github.com/joomla/joomla-cms into versionnumberposition

astridx committed:
header menu mobil

astridx committed:
lint

astridx committed:
correction

astridx commented:
I just added more changes. The top menu in the "footer" has now the following changes (for me improvments):

1. I use the whole free space for showing the icons.
2. If there is not enough space the icons are put into a fly-out menu that you can open with the toggler.
3. The fly-out menu has one column. So it can not happen, that there is much empty space.

I hope I did not miss something again.

infograf768 commented on administrator/language/en-GB/en-GB.mod_version.ini:
```diff
@@ -6,3 +6,4 @@
 MOD_VERSION="Joomla! Version Information"
 MOD_VERSION_XML_DESCRIPTION="This module displays the Joomla! version."
 MOD_VERSION_CURRENT_VERSION_TEXT="The currently installed Joomla! version is \"%s\""
+MOD_VERSION_CURRENT_VERSION_SHORTTEXT="Version: %s"
```
alpha order

infograf768 commented:
I am concerned by the position of the Update Checks module in the Control Panel page. This before and after this patch. But it’s even worse when a unique column is used in smaller screens.

In 3.x we have an alert/message always displaying on top of the page. In 4.0 it is very easy to miss that module as one has to scroll a lot to be informed about updates.

Is there a way (by js as it looks like we can’t avoid these complex js for almost all in J4) to move that entire module to top when there are updates?

bembelimen commented:
Sorry, I fully disagree with this PR.

This PR does not deliver any benefits to the system. It just make the current version less visible + implements a footer where we don't want any footer.

I think we should raise awareness to the installed version + should always point out if the version is out of date.

Additionally if you check the Joomla! 3 version and compare with Joomla! 4 you'll see, that the version button is now an own module, so if it's annoys you, you can disable it with one click.

But version in the footer? No!

brianteeman commented:
maybe there is some other way to make it look less like a button?

chmst commented:
I agree that the appearance of the version module needs to be  improved.  

But do not see any advantage in moving the module to a footer. In modern applications I am used to find all mstatus informations top right not some information on the top, some at the end of the screen. 
It seems not ergonomic to me, the  eye must search for the information above the screen. 
As @bembelimen says, you can move the module at the end of your dashboard.

brianteeman committed:
[4.0] Mail templates (#26471)

Two strings were the wrong way around

Config should say "Allow ..."

Not the individual template

NOTE: I am not happy with any of these language strings but I will do that in a seperate pr

astridx committed:
version number top right

astridx committed:
merge

astridx committed:
correction

astridx commented:
What do you say to this:
![Articles  New   test   Administration(2)](https://user-images.githubusercontent.com/9974686/66256477-7f61d100-e78e-11e9-8ef9-4bb18132fc79.png)
![Articles  New   test   Administration(1)](https://user-images.githubusercontent.com/9974686/66256478-7ffa6780-e78e-11e9-901c-0d7b53a751c8.png)

bembelimen commented:
To make it easy: why not link to the Joomla updater?

N6REJ commented:
why not to the right of the J! icon?  Wouldn't that make the most sense?

astridx commented:
> @N6REJ why not to the right of the J! icon?

because most voted for the top right and because the space on a narrow display is better used. I tried it: 

![Control Panel   test   Administration(5)](https://user-images.githubusercontent.com/9974686/66276408-00a28c00-e893-11e9-97ef-da932e057dee.png)
![Control Panel   test   Administration(4)](https://user-images.githubusercontent.com/9974686/66276409-00a28c00-e893-11e9-9537-942798de57f4.png)

I also put it in the SVG logo as `<text>`. But then it can happen that you can no longer read the numbers.

> @bembelimen To make it easy: why not link to the Joomla updater?

I would be confused if a current version number would link to the updater. I would rather assume a further information about the version or a changelog here.

---

But I also think that there is a misunderstanding or that I do not have all the information. I had asked above, but did not get an answer. At the moment the module does nothing but show the installed version. There is no warning. The description of the module says so too. For a warning regarding an outdated installation, the Quickicon is responsible, right?


![Modules  Joomla  Version Information   test   Administration](https://user-images.githubusercontent.com/9974686/66275711-a4883980-e88b-11e9-9e53-99780a3f4fb6.png)
![Plugins   test   Administration](https://user-images.githubusercontent.com/9974686/66275712-a4883980-e88b-11e9-8c6d-470c4e364296.png)

Quy commented:
How about adding `text-muted` to imply it is not clickable?

![26447](https://user-images.githubusercontent.com/368084/66337049-3bd4a780-e8f3-11e9-8c08-aae6fe706cfb.jpg)

N6REJ commented:
@astridx I don't think you understood what I meant.  Here is what I meant.
![desktop](https://user-images.githubusercontent.com/1850089/66337668-3a13df80-e905-11e9-8482-f9ae47e49ffc.jpg)
![mobile](https://user-images.githubusercontent.com/1850089/66337669-3a13df80-e905-11e9-99de-6750258a5410.jpg)

infograf768 commented:
> For a warning regarding an outdated installation, the Quickicon is responsible, right?

To all: yes it is but please re-read my comment above
https://github.com/joomla/joomla-cms/pull/26447#issuecomment-538615193

N6REJ commented:
@infograf768 since the message modal pops up to top I wouldn't see why it couldn't be used.

infograf768 commented:
> @infograf768 since the message modal pops up to top I wouldn't see why it couldn't be used.

Not sure what you mean.
We have no message when an update (joomla, extensions, overrides, urgent privacy requests) is available. The new Quickicon system does not provide for such messages.
![updatemessage](https://user-images.githubusercontent.com/869724/66457025-9dba1c00-ea6f-11e9-8231-9aaf6e4dfb6f.gif)

To obtain such messages, I guess an important refactoring would be necessary.

wilsonge commented:
https://github.com/joomla/joomla-cms/pull/26734 Added the `text-muted` property proposed by @Quy as it seems like a better way to demonstrate the fact it's not clickable
